src/overlays/actors/ovl_En_Nb/z_en_nb.c

```diff
@@ -21,255 +150,629 @@ const ActorInit En_Nb_InitVars = {
+    } else {
+        this->actor.flags |= 9;
+        maskReaction = Text_GetFaceReaction(globalCtx, 0x23);
+        this->actor.textId = maskReaction;
+        if ((maskReaction & 0xFFFF) == 0) {
+            // "If you can successfully get the Silver Gauntlets...I'll do unk great for you!"
+            this->actor.textId = 0x6026;
+        }
+        func_8002F2F4(&this->actor, globalCtx);
+    }
+}
+
+void func_80AB3B7C(EnNb* this, GlobalContext* globalCtx) {
+    if (func_8010BDBC(&globalCtx->msgCtx) == 2) {
+        this->action = NB_IDLE_AFTER_TALK;
+        this->actor.flags &= ~9;
+    }
+}
+
+void EnNb_WaitForNotice(EnNb* this, GlobalContext* globalCtx) {
+    func_80AB1284(this, globalCtx);
+    EnNb_UpdateCollider(this, globalCtx);
+    EnNb_FrameUpdateMatrix(this);
+    EnNb_UpdateEyes(this);
+    EnNb_CheckNoticed(this, globalCtx);
+}
+
+void EnNb_StandUpAfterNotice(EnNb* this, GlobalContext* globalCtx) {
+    s32 unk;
+
+    func_80AB1284(this, globalCtx);
+    EnNb_UpdateCollider(this, globalCtx);
+    unk = EnNb_FrameUpdateMatrix(this);
+    EnNb_UpdateEyes(this);
+    EnNb_SetupIdleCrawlspace(this, unk);
+}
+
+void EnNb_BlockCrawlspace(EnNb* this, GlobalContext* globalCtx) {
+    func_80AB1284(this, globalCtx);
+    EnNb_UpdateCollider(this, globalCtx);
+    func_80AB0FBC(this, globalCtx);
+    EnNb_FrameUpdateMatrix(this);
+    EnNb_UpdateEyes(this);
+    func_80AB3838(this, globalCtx);
+}
+
+void EnNb_InitCrawlspaceDialogue(EnNb* this, GlobalContext* globalCtx) {
+    func_80AB1284(this, globalCtx);
+    EnNb_UpdateCollider(this, globalCtx);
+    func_80AB0FBC(this, globalCtx);
+    EnNb_FrameUpdateMatrix(this);
+    EnNb_UpdateEyes(this);
+    EnNb_SetTextIdAsChild(this, globalCtx);
+}
+
+void EnNb_FollowPath(EnNb* this, GlobalContext* globalCtx) {
+    s32 unk;
+
+    func_80AB359C(this);
+    func_80AB1284(this, globalCtx);
+    EnNb_UpdateCollider(this, globalCtx);
+    func_80AB36DC(this, globalCtx);
+    func_80AB10C4(this);
+    unk = EnNb_FrameUpdateMatrix(this);
+    EnNb_UpdateEyes(this);
+    func_80AB3A7C(this, globalCtx, unk);
+}
+
+void func_80AB3DB0(EnNb* this, GlobalContext* globalCtx) {
+    func_80AB1284(this, globalCtx);
+    EnNb_UpdateCollider(this, globalCtx);
+    func_80AB0FBC(this, globalCtx);
+    EnNb_FrameUpdateMatrix(this);
+    EnNb_UpdateEyes(this);
+    func_80AB3B04(this, globalCtx);
+}
+
+void func_80AB3E10(EnNb* this, GlobalContext* globalCtx) {
+    func_80AB1284(this, globalCtx);
+    EnNb_UpdateCollider(this, globalCtx);
+    func_80AB1040(this, globalCtx);
+    EnNb_FrameUpdateMatrix(this);
+    EnNb_UpdateEyes(this);
+    func_80AB3B7C(this, globalCtx);
+}
+
+void EnNb_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnNb* this = THIS;
+
+    if (this->action < 0 || this->action > 30 || sActionFuncs[this->action] == NULL) {
+        // "MAIN MODE IS ABNORMAL!!!!!!!!!!!!!!!!!!!!!!!!!"
+        osSyncPrintf(VT_FGCOL(RED) "メインモードがおかしい!!!!!!!!!!!!!!!!!!!!!!!!!\n" VT_RST);
+    } else {
```

Comment:
I'm pretty sure we've been sticking to early returning for error-like cases such as this